File: src/main/java/ch/uzh/ifi/access/projections/TaskSummary.java
package ch.uzh.ifi.access.projections;

import ch.uzh.ifi.access.model.Task;
import ch.uzh.ifi.access.model.TaskInformation;
import org.springframework.data.rest.core.config.Projection;

import java.util.Map;

@Projection(types = {Task.class})
public interface TaskSummary {

    String getSlug();

    Integer getOrdinalNum();

    Double getMaxPoints();

    Map<String,TaskInformation> getInformation();

    Integer getMaxAttempts();

    Integer getAttemptRefill();

    String getDockerImage();

    String getRunCommand();

    String getTestCommand();

    String getGradeCommand();

    Integer getTimeLimit();
}

File: src/main/java/ch/uzh/ifi/access/projections/TaskSummary.kt
package ch.uzh.ifi.access.projections

import ch.uzh.ifi.access.model.Task
import ch.uzh.ifi.access.model.TaskInformation
import org.springframework.data.rest.core.config.Projection

@Projection(types = [Task::class])
interface TaskSummary {
    val slug: String?
    val ordinalNum: Int?
    val maxPoints: Double?
    val information: Map<String?, TaskInformation?>?
    val maxAttempts: Int?
    val attemptRefill: Int?
    val dockerImage: String?
    val runCommand: String?
    val testCommand: String?
    val gradeCommand: String?
    val timeLimit: Int?
}